vllm/adapter_commons/worker_manager.py

```diff
@@ -0,0 +1,63 @@
+from abc import ABC, abstractmethod
+from typing import Any, Optional, Set
+
+import torch
+
+
+class AbstractWorkerManager(ABC):
+
+    def __init__(self, device: torch.device):
+        self.device = device
+
+    @property
+    @abstractmethod
+    def is_enabled(self) -> bool:
+        ...
+
+    def set_active_adapters(self, requests: Set[Any],
+                            mapping: Optional[Any]) -> None:
+        self._apply_adapters(requests)
+        self._adapter_manager.set_adapter_mapping(mapping)
+
+    @abstractmethod
+    def _load_adapter(self, request: Any) -> Any:
```

Comment:
likewise, let's make public methods abstract instead of private ones and push the implementation to subclasses (we can reuse code through functions)